java-checks/src/test/files/checks/UnnecessarySemicolonCheck.java

```diff
@@ -0,0 +1,29 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class UnnecessarySemicolonCheck {
+
+  private static InputStream input(String path) throws IOException {
+    return Files.newInputStream(Paths.get(path));
+  }
+
+  public static void main(String[] args) throws IOException {
+
+    try(InputStream i1 = input("i1");) { // Noncompliant {{Remove this extraneous semicolon.}}
+    }
+
+    try(InputStream i1 = input("i1");
```

Comment:
For this test case, put the two input streams on the same line and use precise issue location testing, it will help guarantee that we are only raising issue on the last semi-colon.

You can use precise issue location like this:
```// Noncompliant [[sc=14;ec=15]]```